src/game.rs

```diff
@@ -56,14 +57,68 @@ impl PlayerIdGenerator {
     }
 }
 
+/// Generator for a player username.
+///
+#[derive(Debug)]
+pub struct PlayerNameGenerator;
+
+impl PlayerNameGenerator {
+
+    pub fn generate_username(&self) -> String {
+
+        let name_list = vec!
+        [
+            "Hiphopopotamus",
+            "Rhymenocerous",
+            "Steve",
+            "Peter Potamus",
+            "Mr. Wiggles",
+            "Seargent Snout",
+            "Calamity Hippopatamy",
+            "Ringo Potamus",
+            "Mrs. Basil E. Frankenhippo",
+            "Harry Pottamus",
+            "Hermoine Potamus",
+            "Buckbeak",
+            "Marie Hippolonium",
+            "Hippope Francis",
+            "Hippaul Potamus",
+            "Danerys Mother of Hippos",
+            "Darth Potamus",
+            "Ann Perkopotamins!",
+            "Hippopotahut",
+            "Hippopotabell",
+            "Combination Hippopotahut and Hippopotabell",
+            "Hippchat",
+            "Zippo",
+            "Let 'er Rippo",
+            "Have a Nice Trippo",
+            "Tortilla Chippo",
+            "Lastey",
+            "Jean-Baptiste Emanuel Hippo",
+            "Hippo Hipposon",
+            "Son of Potamus",
+        ];
+
+        match thread_rng().choose(&name_list) {
+            None => "".to_string(),
+            Some(name) => name.to_string(),
+        }
```

Comment:
[`Rng::choose`](https://docs.rs/rand/0.3.15/rand/trait.Rng.html#method.choose) only returns `None` if `name_list` is empty. In this case, we know for sure that `name_list` isn't empty, so it would be better to simply do:

```
thread_rng().choose(&name_list).unwrap()
```